src/test/regress/expected/foreign_key_to_reference_table.out

```diff
@@ -0,0 +1,986 @@
+-- ON UPDATE CASCADE and see if it works properly with cascading upsert
+CREATE TABLE referenced_table(test_column int, test_column2 int, PRIMARY KEY(test_column));
+CREATE TABLE referencing_table(id int, ref_id int DEFAULT -1);
+SELECT create_reference_table('referenced_table');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+SELECT create_distributed_table('referencing_table', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+ALTER TABLE referencing_table ADD CONSTRAINT fkey_ref FOREIGN KEY (ref_id) REFERENCES referenced_table(test_column) ON UPDATE CASCADE;
+INSERT INTO referenced_table SELECT x, x FROM generate_series(0,1000) AS f(x);
+INSERT INTO referencing_table SELECT x, x FROM generate_series(0,1000) AS f(x);
+INSERT INTO referenced_table VALUES (1,2), (2,3), (3,4), (4,5)
+ON CONFLICT (test_column)
+DO UPDATE
+  SET test_column = -1 * EXCLUDED.test_column;
+SELECT * FROM referencing_table WHERE ref_id < 0 ORDER BY 1;
+ id | ref_id 
+----+--------
+  1 |     -1
+  2 |     -2
+  3 |     -3
+  4 |     -4
+(4 rows)
+
+DROP TABLE referenced_table CASCADE;
+NOTICE:  drop cascades to constraint fkey_ref on table referencing_table
+DROP TABLE referencing_table CASCADE;
+-- Chained references 
+-- In the following test, we create foreign keys from one column in a distributed
+-- table to two reference tables. We expect to see that even if a data exist in
+-- one reference table, it is not going to be inserted in to referencing table
+-- because of lack of the key in the other table. Data can only be inserted into 
+-- referencing table if it exists in both referenced tables.
+-- Additionally, delete or update in one referenced table should cascade properly.
+CREATE TABLE referenced_table(test_column int, test_column2 int, PRIMARY KEY(test_column));
+CREATE TABLE referenced_table2(test_column int, test_column2 int, PRIMARY KEY(test_column2));
+CREATE TABLE referencing_table(id int, ref_id int);
+SELECT create_reference_table('referenced_table');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+SELECT create_reference_table('referenced_table2');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+SELECT create_distributed_table('referencing_table', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+ALTER TABLE referencing_table ADD CONSTRAINT fkey_ref FOREIGN KEY (id) REFERENCES referenced_table(test_column) ON DELETE CASCADE;
+ALTER TABLE referencing_table ADD CONSTRAINT fkey_ref2 FOREIGN KEY (id) REFERENCES referenced_table2(test_column2) ON DELETE CASCADE;
+SELECT * FROM table_fkeys_in_workers WHERE relid LIKE 'fkey_reference_table.%' AND refd_relid LIKE 'fkey_reference_table.%' ORDER BY 1,2,3;
+       name        |                     relid                      |                   refd_relid                   
+-------------------+------------------------------------------------+------------------------------------------------
```

Comment:
Hmm, the order of the output of this test and the following test in windows is different. It is very likely that it is because of different `LC_COLLATE ` during `initdb`. It seems like the order of `fkey_ref_` and `fkey_ref2_` are not consistent in these systems.

A quick thing to test would change `fkey_ref2` to something like `foreign_key_2` such that it is different than `fkey_ref`.